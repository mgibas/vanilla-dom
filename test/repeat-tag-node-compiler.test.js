const compiler = require('../src/repeat-tag-node-compiler')
let tagNodeCompiler = require('../src/tag-node-compiler')
tagNodeCompiler.compile = jest.fn()
tagNodeCompiler.compile.mockImplementation((a,b,c) => `TAG_NODE_COMPILER_PART('${a}', '${b}', '${c}')`)

describe('repeat-tag-node-compiler', () => {
  describe('on compile', () => {
    it('simple repeat-for', () => {
      expect(compiler.compile('parent', {
        attribs: { 'repeat-for': '{{state.objects}}' }
      }, (child, compiler) => {}))
        .toMatchSnapshot();
    })
  })
})
