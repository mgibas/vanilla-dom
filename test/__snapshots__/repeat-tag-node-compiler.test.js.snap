// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repeat-tag-node-compiler on compile simple repeat-for 1`] = `
"(() => {
  let items = (() => {
    try {
      return state.objects;
    } catch (err) {
      return;
    }
  })();
  if (!items)
    return;
  for (let i = 0; i < items.length; i++) {
    TAG_NODE_COMPILER_PART('parent', '[object Object]', '(child, compiler) => {}');
  }
})();"
`;
